pkg/logging/yaml_parser.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package logging
+
+type ReqLogConfig struct {
+	HTTP    ProtocolConfigs `yaml:"http"`
+	GRPC    ProtocolConfigs `yaml:"grpc"`
+	Options OptionsConfig   `yaml:"options"`
+}
+
+type ProtocolConfigs struct {
+	Options OptionsConfig    `yaml:"options"`
+	Config  []ProtocolConfig `yaml:"config"`
+}
+
+type OptionsConfig struct {
+	Level    string         `yaml:"level"`
+	Decision DecisionConfig `yaml:"decision"`
+}
+
+type DecisionConfig struct {
+	LogStart bool `yaml:"log_start"`
+	LogEnd   bool `yaml:"log_end"`
+}
+
+type ProtocolConfig struct {
+	// The first two options are for HTTP config.
+	Path string `yaml:"path"`
```

Comment:
I don't think it makes sense as we have too many optional fields - it's extremely hard to tell how it use (:

Reply:
Resolved: https://github.com/thanos-io/thanos/pull/3361/files#r545626611